.github/workflows/test.yml

```diff
@@ -0,0 +1,61 @@
+name: 🐘 Test
+on:
+  push:
+  pull_request:
+  schedule:
+    - cron:  '0 14 3 * *' # Monthly at 2pm on the third
+jobs:
+  build:
+    strategy:
+      matrix:
+        pg:
+          - { version: 14,  upgrade_to: "",  update_from: 0.99.0 }
+          - { version: 13,  upgrade_to: 14,  update_from: 0.99.0 }
+          - { version: 12,  upgrade_to: 13,  update_from: 0.99.0 }
+          - { version: 11,  upgrade_to: 12,  update_from: 0.99.0 } # Versions prior to 0.99.0 don't support Postgres 11
+          - { version: 10,  upgrade_to: 11,  update_from: 0.95.0 }
+          - { version: 9.6, upgrade_to: 10,  update_from: 0.95.0 }
+          - { version: 9.5, upgrade_to: 9.6, update_from: 0.95.0 }
+          - { version: 9.4, upgrade_to: 9.5, update_from: 0.95.0 }
+          - { version: 9.3, upgrade_to: 9.4, update_from: 0.95.0 }
+          - { version: 9.2, upgrade_to: 9.3, update_from: "" }     # updatecheck is not supported prior to 9.3
+          - { version: 9.1, upgrade_to: 9.2, update_from: "" }
+          # Also test pg_upgrade across many versions
+          # - { version: 9.1,  upgrade_to: 14, update_from: 0.99.0 }
+          # - { version: 9.4,  upgrade_to: 14, update_from: 0.99.0 }
+        parallel:
+          - { conn: "",  name: 'parallel' }
+          - { conn: "1", name: 'serial'   }
+    name: 🐘 PostgreSQL ${{ matrix.pg.version }} (${{ matrix.parallel.name }})
+    runs-on: ubuntu-latest
+    container: pgxn/pgxn-tools
+    env:
+      PGUSER: postgres
+      PARALLEL_CONN: "${{ matrix.parallel.conn }}"
+      UPDATE_FROM:  "${{ matrix.pg.update_from }}"
+    steps:
+      - run: pg-start ${{ matrix.pg.version }}
+      - uses: actions/checkout@v2
+
+      # Basic regression test.
+      - run: pg-build-test
+
+      # Test update.
+      - run: 'if [ -d "$UPDATE_FROM" ]; then make uninstall clean updatecheck; fi'
+
+      # Test upgrade.
+      # - run: ./test/test_MVU.sh -s 55667 55778 "{{ matrix.pg.version }}" "{{ matrix.pg.upgrade_to }}" "/usr/lib/postgresql/{{ matrix.pg.version }}/bin/" "/usr/lib/postgresql/{{ matrix.pg.upgrade_to }}/bin/"
+
+      # Test all, install, test, test-serial, and test-parallel, both from clean
+      # repo and repeated with existing build.
+      - run: make uninstall clean all
```

Comment:
Given the number of commands here (and especially since there's some conditional stuff that we probably want) maybe worth splitting this out to a shell script...

Reply:
I like to have them here because then they're each reported separately in the Actions display. Here's [an example](https://github.com/theory/pgtap/runs/4127656711?check_suite_focus=true).